src/transport/builder/umessagebuilder.rs

```diff
@@ -0,0 +1,733 @@
+    /// assert_eq!(message.attributes.priority, UPriority::UPRIORITY_CS1.into());
+    /// assert_eq!(message.attributes.source, Some(reply_to_address).into());
+    /// assert_eq!(message.attributes.sink, Some(method_to_invoke).into());
+    /// assert_eq!(message.attributes.ttl, Some(5000));
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn request(
+        method_to_invoke: &'a UUri,
+        reply_to_address: &'a UUri,
+        ttl: u32,
+    ) -> UMessageBuilder<'a> {
+        UMessageBuilder {
+            validator: Box::new(crate::transport::validator::RequestValidator),
+            message_type: UMessageType::UMESSAGE_TYPE_REQUEST,
+            source: Some(reply_to_address),
+            sink: Some(method_to_invoke),
+            ttl: Some(i32::try_from(ttl).unwrap_or(i32::MAX)),
+            ..Default::default()
+        }
+    }
+
+    /// Gets a builder for creating an RPC *response* message.
+    ///
+    /// A response message is used to send the outcome of processing a request message
+    /// to the original sender of the request message.
+    ///
+    /// # Arguments
+    ///
+    /// * `reply_to_address` - The URI that the sender of the request expects to receive the response message at.
+    /// * `request_id` - The identifier of the request that this is the response to.
+    /// * `invoked_method` - The URI identifying the method that has been invoked and which the created message is
+    ///                      the outcome of.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use up_rust::uprotocol::SerializationError;
+    /// use up_rust::uuid::builder::UUIDBuilder;
+    /// use up_rust::uprotocol::{UMessageType, UPriority, UUri};
+    /// use up_rust::transport::builder::UMessageBuilder;
+    ///
+    /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+    /// let invoked_method = UUri::try_from("my-vehicle/cabin/1/rpc.unlock_doors")?;
+    /// let reply_to_address = UUri::try_from("my-cloud/dashboard/1/rpc.response")?;
+    /// let request_id = UUIDBuilder::new().build();
+    /// let message = UMessageBuilder::response(&reply_to_address, &request_id, &invoked_method).build()?;
+    /// assert_eq!(message.attributes.type_, UMessageType::UMESSAGE_TYPE_RESPONSE.into());
+    /// assert_eq!(message.attributes.priority, UPriority::UPRIORITY_CS1.into());
+    /// assert_eq!(message.attributes.source, Some(invoked_method).into());
+    /// assert_eq!(message.attributes.sink, Some(reply_to_address).into());
+    /// assert_eq!(message.attributes.reqid, Some(request_id).into());
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn response(
+        reply_to_address: &'a UUri,
+        request_id: &'a UUID,
+        invoked_method: &'a UUri,
+    ) -> UMessageBuilder<'a> {
+        UMessageBuilder {
+            validator: Box::new(crate::transport::validator::ResponseValidator),
+            message_type: UMessageType::UMESSAGE_TYPE_RESPONSE,
+            source: Some(invoked_method),
```

Comment:
Also related to the [issue I made](https://github.com/eclipse-uprotocol/up-spec/issues/68).